scipy/stats/_continuous_distns.py

```diff
@@ -3569,34 +3569,63 @@ def fit(self, data, *args, **kwds):
         data, floc, fscale = _check_fit_input_parameters(self, data,
                                                          args, kwds)
 
-        # if user has provided `floc` or `fscale`, fall back on super fit
-        # method. This scenario is not suitable for solving a system of
-        # equations
-        if floc is not None or fscale is not None:
-            return super().fit(data, *args, **kwds)
-
-        # rv_continuous provided guesses
-        loc, scale = self._fitstart(data)
-        # account for user provided guesses
-        loc = kwds.pop('loc', loc)
-        scale = kwds.pop('scale', scale)
-
         # By the method of maximum likelihood, the estimators of the
-        # location and scale are the roots of the equation defined in
+        # location and scale are the roots of the equations defined in
         # `func` and the value of the expression for `loc` that follows.
-        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
-        # and Peacock (2000), Page 101
+        # The first `func` is a first order derivative of the log-likelihood
+        # equation and the second is from Source: Statistical Distributions,
+        # 3rd Edition. Evans, Hastings, and Peacock (2000), Page 101.
+
+        def get_loc_from_scale(scale):
+            return -scale * (sc.logsumexp(-data / scale) - np.log(len(data)))
+
+        if fscale is not None:
+            # if the scale is fixed, the location can be analytically
+            # determined.
+            scale = fscale
+            loc = get_loc_from_scale(scale)
+
+        else:
+            # A different function is solved depending on whether the location
+            # is fixed.
+            if floc is not None:
+                loc = floc
+
+                # equation to use if the location is fixed.
+                def func(scale):
+                    term1 = (loc - data) * np.exp((loc - data) / scale) + data
```

Comment:
apart from that, I could follow the calculation

Reply:
what I just realized: if loc is fixed, we cannot use the first equation for scale in book "Statistical distributions". this is because the first equation is derived by using the MLE condition for loc (i.e., loc has been replaced  by an expression involving scale only, the inverse of get_loc_from_scale). but if loc is fixed, the derivate w.r.t. loc cannot be assumed to be zero, so we cannot do that. maybe it is worth adding a comment?

Reply:
Yes, I think that's right. That's what I was referring to below that:
>  @swallan IIRC you confirmed that the equations we're using for fixed location or scale simplify to these when both are free? 

We'll see how well memory serves, though. Either way, yes, a comment sounds helpful.

Reply:
i made a proposal